### PR TITLE
Luminex Levey-Jennings report grid improvements

### LINKED ISSUE #573: DHVI - Luminex Levey-Jennings Report Improvements - Levey-Jennings data grid improvements dev
[Epic](https://github.com/LabKey/commonAssays/issues/571)
[Spec](https://docs.google.com/document/d/1IJIHAfxJ5wdMCOkrNYcON8aFbiC1za34wuQhCPyh11k/edit)

**Tasks**

- [x] Switch L-J report grid to LabKey data region - 5hr
- [x] Allow for grid view customization - 3hr
- [x] Sync grid filtering with plot display - 3hr
- [x] Handle both the titration and single point control version of the report - 3hr
- [x] Highlight grid Guide Set Start Date column values to be bold if members of guide set - 2hr
- [x] Reapply “Apply Guide Set” button functionality - 3hr
- [x] Reapply “View 4PL Curves” button functionality - 2hr
- [x] Reapply “QC Flags” column functionality - 2hr
- [x] Remove current filter options that are above the plot - 0.5hr
- [x] Update spec - 0.5hr
- [x] Code review - 1hr @labkey-klum 
- [x] Test plan - 1hr
- [x] Manual testing - 2hr @labkey-tchad 
- [x] Selenium test updates - 3hr @cnathe 
- [x] TeamCity review and merge - 1hr

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
The data grid on the Luminex Levey-Jennings report uses an Ext grid panel which limits some of the abilities that we normally see in LabKey grids, most notably the ability to page through results, customize the grid columns, and filter on arbitrary columns in the grid. This PR converts that grid to a LabKey DataRegion. This allows for the features mentioned and retains the ability to apply guide sets to data grid rows and view 4PL curve fits (for titration based reports).

#### Related Pull Requests
* https://github.com/LabKey/commonAssays/pull/588

#### Changes
* convert tracking data grid from Ext4 grid panel to LK QueryWebPart
* plot data query split to get metric data and guide set data separately
* add back "Apply Guide Set" button to the dataRegion buttonBar header
* add back "View 4PL Curves" button to the dataRegion buttonBar header
* data grid styling fix to remove extContainer styles from wrapping component

### CODE REVIEW

**Comment by cnathe** [comment]
@labkey-klum Karl, would you mind taking the code review for this PR? Most of the changes are converting from the Ext4 grid to the LabKey DataRegion (via QueryWebpart) on the Luminex Levey-Jennings report page. We can sync up and I can demo the changes for you before you dive in.

File: luminex/test/src/org/labkey/test/util/luminex/LuminexGuideSetHelper.java [hunk 1]
```diff
@@ -89,6 +89,13 @@ public void verifyGuideSetsNotApplied(String assayName)
         table.clearFilter("GuideSet/Created");
     }
 
+    public DataRegionTable getTrackingDataRegion()
```
Comment (labkey-klum) [inline luminex/test/src/org/labkey/test/util/luminex/LuminexGuideSetHelper.java:92]: Is this just a helper for the tests?

Comment (labkey-klum) [reply luminex/test/src/org/labkey/test/util/luminex/LuminexGuideSetHelper.java:92]: Oh nvm, I didn't notice the package.

File: luminex/webapp/luminex/LeveyJenningsTrackingDataPanel.js [hunk 2]
```diff
@@ -28,520 +23,438 @@ LABKEY.LeveyJenningsTrackingDataPanel = Ext.extend(Ext.grid.GridPanel, {
...
-        var storeConfig = {
-            assayName: this.assayName,
-            controlName: this.controlName,
-            controlType: this.controlType,
-            analyte: this.analyte,
-            isotype: this.isotype,
-            conjugate: this.conjugate,
-            maxRows: !hasReportFilter ? this.defaultRowSize : undefined,
+        this.qwp = new LABKEY.QueryWebPart({
+            renderTo: 'trackingDataPanel_QWP',
```
Comment (labkey-klum) [inline luminex/webapp/luminex/LeveyJenningsTrackingDataPanel.js:128]: There will never be more than a single view per page?

**Review by labkey-klum — APPROVED** [review]
Overall looks great and nice to migrate off of the `LABKEY.editorGridPanel`. Did we consider whether it made sense to migrate off of Ext3?

File: luminex/webapp/luminex/LeveyJenningsTrackingDataPanel.js (hunk 2 above)
Comment (cnathe) [reply luminex/webapp/luminex/LeveyJenningsTrackingDataPanel.js:128]: Correct. There is just one data grid on the report page. Any change to the filters or graph parameters will replace the QWP here.

**Comment by cnathe** [comment]
> Overall looks great and nice to migrate off of the `LABKEY.editorGridPanel`. Did we consider whether it made sense to migrate off of Ext3?

The client was pretty budget conscious, so we did not consider including that here. I'd be all for migrating the LJ report off of Ext3 (and off of Ext all together!), but I don't have a good estimate for that at this time.

**Pull request merged**